#!/usr/bin/env python
PACKAGE = "pointcloud_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_crop_height",     double_t,    0,     "Minimum crop height for the box filter",          1,    -20,  20)
gen.add("max_crop_height",    double_t,    0,     "Maximum crop height for the box filter",          1,    -20,   20)
gen.add("outlier_filter_mean", double_t,    0,     "Outlier filter mean",          1,    0,   200)
gen.add("outlier_filter_stddev", double_t,    0,     "Outlier filter stddev",          1,    0,   20)
gen.add("dilation_factor", double_t,    0,     "dilation_factor",          1,    0,   20)
gen.add("min_match_size", double_t,    0,     "min_match_size",          10,    0,   10000)

exit(gen.generate(PACKAGE, "pointcloud_filter", "WallDetectionParameters"))